controllers/node.go

```diff
@@ -1064,24 +1064,34 @@ func deleteNode(w http.ResponseWriter, r *http.Request) {
 		// check if server should be removed from mq
 		// err is irrelevent
 		nodes, _ := logic.GetAllNodes()
-		var foundNode models.Node
+		var foundNode *models.Node
 		for _, nodetocheck := range nodes {
 			if nodetocheck.HostID == node.HostID {
-				foundNode = nodetocheck
+				foundNode = &nodetocheck
 				break
 			}
 		}
+		if foundNode == nil { // check if node is in deleted nodes if not found
+			delNode, err := logic.GetDeletedNodeByID(node.ID.String())
```

Comment:
Node deletion already happens at line 1053, i have fixed this already in my current PR #1905

Reply:
this is the case in which a node receives a DELETE action and responds itself rather the API